### PR TITLE
Added code and tests

### PR DESCRIPTION
**Pull request by tavisca-sjana**

### CODE REVIEW

File: CartApp/CartItem.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿namespace CartApp
+{
+    public class CartItem 
+    {
+        public int Quantity;
```
Comment (tavisca-pkadam) [inline CartApp/CartItem.cs:5]: member field camel Casing

File: CartApp/CartItem.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        public readonly Product Product;
+        public Discount Discount = new Discount();
```
Comment (tavisca-pkadam) [inline CartApp/CartItem.cs:7]: Discount naming does'nt suggest much.... is it discount percentage ... or on amount

File: CartApp.sln [hunk 3]
```diff
@@ -0,0 +1,31 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 15
```
Comment (tavisca-snarhe) [inline CartApp.sln:3]: Please add .gitignore file in repo

File: CartApp/Cart.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+
+namespace CartApp
+{
+    public class Cart
+    {
+        private List<CartItem> _cartItem = new List<CartItem>();
```
Comment (tavisca-snarhe) [inline CartApp/Cart.cs:7]: Assignment should be in constructor

File: CartApp/Cart.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        {
+            _cartItem.Add(item);
+        }
+
+        public void Remove(CartItem item)
+        {
+            _cartItem.Remove(item);
+        }
+
+        public List<CartItem> ViewItems()
```
Comment (tavisca-snarhe) [inline CartApp/Cart.cs]: If you are returning from this function it should be named like GetItems

File: CartApp/Program.cs [hunk 6]
```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace CartApp
+{
+    class Program
```
Comment (KumarWaghmode) [inline CartApp/Program.cs:6]: Move classes in different files

File: CartApp/Program.cs [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        dairy = 10,
+        educational = 20,
+        entertainment = 5,
+        electronics = 8
+    }
+
+    public class DiscountFactory
+    {
+        public int GenerateDiscount(string category)
```
Comment (KumarWaghmode) [inline CartApp/Program.cs]: Method name should be GetDiscount

**Review by KumarWaghmode — CHANGES_REQUESTED** [review]
Discount strategy implementation is missing

File: CartApp/CartItem.cs (hunk 1 above)
Comment (tavisca-sjana) [reply CartApp/CartItem.cs:5]: I guess public field should be in PascalCase.

**Pull request merged**